modules/katello_manifest.py

```diff
@@ -0,0 +1,182 @@
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: katello_manifest
+short_description: Create and Manage Katello manifests
+description:
+    - Create and Manage Katello manifests
+author: "Andrew Kofink (@akofink)"
+requirements:
+    - "nailgun >= 0.29.0"
+    - "python >= 2.6"
+options:
+    server_url:
+        description:
+            - URL of Foreman server
+        required: true
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    verify_ssl:
+        description:
+            - Verify SSL of the Foreman server
+    organization:
+        description:
+            - Organization that the sync plan is in
+        required: true
+    manifest_path:
+        description:
+            - Path to the manifest zip file
+        required: true
+    state:
+        description:
+            - The state of the manifest
+        required: true
+        choices:
+            - absent
+            - present
+    repository_url:
+        description:
+            - URL to retrieve content from
+        default: https://cdn.redhat.com
+'''
+# HERE
+EXAMPLES = '''
+- name: "Upload the RHEL developer edition manifest"
+  katello_manifest:
+    username: "admin"
+    password: "admin"
+    server_url: "https://katello.example.com"
+    verify_ssl: false
+    organization: "Default Organization"
+    state: present
+    manifest_path: "/tmp/manifest.zip"
+  delegate_to: localhost
+'''
+
+RETURN = '''# '''
+
+try:
+    from nailgun.entities import (
+        Subscription
+    )
+
+    from nailgun.entity_mixins import (
+        TaskFailedError
+    )
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+from ansible.module_utils.ansible_nailgun_cement import (
+    create_server,
+    ping_server,
+    find_organization,
+    current_subscription_manifest,
+)
+
+
+def validate_params(module, state=None, manifest_path=None):
+    manifest_supplied = not (manifest_path is None or len(manifest_path) == 0)
+    if state == 'present' and not manifest_supplied:
+        module.fail_json(msg="manifest_path is required when state is present.")
+    elif state != 'present' and manifest_supplied:
```

Comment:
I think its better to be explicit here by checking for `== 'absent'` since that is a valid state to be in and prevents future breakage if you add more states.